src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -348,18 +348,92 @@ def create_managed_disk(cmd, resource_group_name, disk_name, location=None,  # p
                                                              disk_sku)
-                client = _compute_client_factory(cmd.cli_ctx)
-                response = client.virtual_machine_images.get(location, disk_publisher, disk_offer, disk_sku,
-                                                             disk_version)
-                image_reference = response.id
             else:  # error
                 raise CLIError('usage error: --image-reference should be ID or URN (publisher:offer:sku:version).')
+        else:
+            from msrestazure.tools import parse_resource_id
+            terms = parse_resource_id(image_reference)
+            disk_publisher, disk_offer, disk_sku, disk_version = \
+                terms['child_name_1'], terms['child_name_3'], terms['child_name_4'], terms['child_name_5']
+
+        client = _compute_client_factory(cmd.cli_ctx)
+        response = client.virtual_machine_images.get(location, disk_publisher, disk_offer, disk_sku,
+                                                     disk_version)
+
+        if hasattr(response, 'hyper_v_generation'):
+            if response.hyper_v_generation == 'V1':
+                logger.warning('Please consider upgrading security for your VM resources by using Gen 2 OS image and '
+                               'Trusted Launch security type. To know more about Trusted Launch, please visit '
+                               'https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch')
+            elif hyper_v_generation and hyper_v_generation == 'V1':
+                logger.warning('Please consider upgrading security for your VM resources by using Gen 2 OS Disk and '
+                               'Trusted Launch security type. Please use "--hyper-v-generation V2" to set the '
+                               'hypervisor generation as Gen2 for OS disk. To know more about Trusted Launch, '
+                               'please visit https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch')
+            elif response.hyper_v_generation == 'V2':
+                # will set default value of hyper_v_generation
+                if hasattr(response, 'features') and response.features \
+                        and 'SecurityType' in response.features \
+                        and response.features['SecurityType'] == 'TrustedLaunchSupported':
+                    # will set default value of security_type
+                    logger.warning('Starting Build 2023 event, az disk create command will deploy Trusted Launch VM '
+                                   'by default. Please use "--security-type TrustedLaunch" in your scripts in advance '
+                                   'to avoid the breaking change. To know more about Trusted Launch, please visit '
+                                   'https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch')
```

Comment:
```suggestion
            elif response.hyper_v_generation == 'V2':
                if and hyper_v_generation == 'V1':
                      logger.warning('Please consider upgrading security for your VM resources by using Gen 2 OS Disk and '
                                     'Trusted Launch security type. Please use "--hyper-v-generation V2" to set the '
                                     'hypervisor generation as Gen2 for OS disk. To know more about Trusted Launch, '
                                     'please visit https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch')
                # will set default value of hyper_v_generation
                if hasattr(response, 'features') and response.features \
                        and 'SecurityType' in response.features \
                        and response.features['SecurityType'] == 'TrustedLaunchSupported':
                    # will set default value of security_type
                    logger.warning('Starting Build 2023 event, az disk create command will deploy Trusted Launch VM '
                                   'by default. Please use "--security-type TrustedLaunch" in your scripts in advance '
                                   'to avoid the breaking change. To know more about Trusted Launch, please visit '
                                   'https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch')
```